### ISSUE TITLE
RCS1047 Doesn't respect generic constraints

### ISSUE THREAD

**Issue by wekempf** [JosefPihrt/Roslynator#316]
I've got a generic type which constrains its type argument to a Task, but a method that returns that type is flagged by RCS1047 for using the Async suffix.

```csharp
public interface IAsyncCommand<in TRequest, out TResponse>
   where TResponse : Task
{
   TResponse ExecuteAsync(TRequest request);
}
```

Due to the constraint, this is a Task returning method and should have the Async suffix, so the analyzer in this case should not be issuing a diagnostic.
